src/cdk/testing/harness-environment.ts

```diff
@@ -12,10 +12,20 @@ import {
   ComponentHarnessConstructor,
   HarnessLoader,
   HarnessPredicate,
-  LocatorFactory
+  HarnessQuery,
+  LocatorFactory,
+  LocatorFnResult,
 } from './component-harness';
 import {TestElement} from './test-element';
 
+/** Parsed form of the queries passed to the `locatorFor*` methods. */
+type ParsedQueries<T extends ComponentHarness> = {
+  allQueries: (string | HarnessPredicate<T>)[],
```

Comment:
Would it make sense to add comments here, like `allQueries` is ordered as specified by input?

and for `harnessTypes` that it is deduped, since multiple predicates could refer to the same harness type?